doc/manuals/name-encoding.md

```diff
@@ -0,0 +1,47 @@
+# Database and collection name encoding and decoding
+
+MongoDB imposes a set of restrictions regarding the characters which cannot be used as part of the databases and collection names, as well as other limitations. These restrictions are detailed at https://docs.mongodb.com/manual/reference/limits/
+
+Since the names of the databases and the collections used by the STH component are composed from the concatenation of the values assigned to the service, service path, entity id, entity type, attribute name and attribute type (depending on the chosen data model) and we do not want the potential users of the STH component to inherit the restrictions imposed by MongoDB (as the database used by the STH component), a codification mechanism is used before conforming the name of the databases and collections.
+
+The codification mechanism is detailed next:
+    1. For database names:
+        1. Encode the forbidden characters (i.e., `/``\``.``(blank space)``"``$``(null character)`) using an escaping character (`x`) and the numerical Unicode code for each character. For instance, the `$` character will be encoded as `x0024`.
+        2. User defined strings already using the above encoding must be escaped prepending another `x`, for instance: the text `x002a` will be encoded as `xx002a`.
+        3. To avoid the case sensitivity restrictions imposed by MongoDB to the database names, uppercase characters (since they are supposed to be less common than lowercase ones) are also be encoded (i.e., `Service` -> `x0053ervice`).
+        4. The name separator character (`xffff`) is not decoded.
```

Comment:
It is NOT "decoded" since it corresponds to the separator character, I mean, for a collection-per-entity model, if the db prefix is `sth_`, the collection prefix is `sth_`, the service is `thexffffservice`, the service path is `/thexffffservicepath`, the entity id is `entityId` and the entity type is `entityType`: the database name would be `thexxffffservice`(the `xffff` group is encoded in case it is present in the service) and the collection name would be `x002fthexxffffservicepathxffffentityIdxffffentityType`... And consequently to decode the elements, the separator (the `xffff` group) would NOT be decoded but interpreted as the separator (if the `xffff` group appears in an encoded name, it is the separator): `x002fthexxffffservicepath` + separator + `entityId` + `entityType`.

Anyhow, I admit it is somehow misleading. Maybe, I could change the sentence to: `4. The separator used to concatenate the name elements is the character group `xffff`. If this group of characters appears in an encoded database or collection name it univocally corresponds to the separator character since in case of appearing in any of the elements it would be encoded as xxffff.`

Anyhow, open to suggestions... :)